apps/examples/board_specific_demo/ameba_secure_storage_demo/secure_storage.c

```diff
@@ -63,88 +66,160 @@ int sstorage_main(int argc, char *argv[])
 	security_handle hnd;
 	security_error res;
 	security_data data;
-	uint32_t slot;
+	uint32_t input_data;
+	uint32_t slot_index;
 	uint32_t address;
 	uint32_t length = TEST_DATA_LENGTH;
 	char ss_path[7];
+	int option;
 
 	printf("Start testing SE Secure Storage!!\n");
+
+	while ((option = getopt(argc, argv, "d:s:h")) > 0) {
+		switch (option) {
+		case 'd':
+			sscanf(optarg, "%x", &input_data);
+			break;
+		case 's':
+			slot_index = atoi(optarg);
+			if (slot_index > 31) {
+				slot_index = 0;
+			}
+			break;
+		case 'h':
+			printf("Usage command: -\n sstorage -s <slot index> -d <hex data value>\n");
+			return 0;
+		default:
+			break;
+		}
+	}
+
+	if (argc < 4) {
```

Comment:
Should we not check this before trying to extract the arguments? The above while loop should then be inside the else part.

Reply:
Modified